Stack/Opc.Ua.Core/Types/Encoders/XmlDecoder.cs

```diff
@@ -1074,7 +1074,7 @@ public ExpandedNodeId ReadExpandedNodeId(string fieldName)
                 EndField(fieldName);
             }
 
-            if (m_namespaceMappings != null && m_namespaceMappings.Length > value.NamespaceIndex)
+            if (m_namespaceMappings != null && m_namespaceMappings.Length > value.NamespaceIndex && !value.IsNull)
             {
```

Comment:
should the same check be applied to the test for SetServerIndex below at line 1082?